tasks/fork/create-checkpoints.ts

```diff
@@ -0,0 +1,109 @@
+import { task, types } from "hardhat/config";
+import { Address } from "viem";
+import {
+    HyperdriveDeployBaseTask,
+    HyperdriveDeployBaseTaskParams,
+} from "../deploy";
+import { sleep } from "./lib";
+
+export type CreateCheckpointsParams = HyperdriveDeployBaseTaskParams & {
+    pollInterval: number;
+};
+
+export function getCheckpointId(
+    checkpointDuration: bigint,
+    blockTimestamp: bigint,
+) {
+    return blockTimestamp - (blockTimestamp % checkpointDuration);
+}
+
+HyperdriveDeployBaseTask(
+    task(
+        "fork:create-checkpoints",
+        "Long-running process that creates checkpoints for all hyperdrive instances when necessary",
+    ),
+)
+    .addOptionalParam(
+        "pollInterval",
+        "Amount of time in minutes between polls for checkpoint state",
+        240,
+        types.int,
+    )
```

Comment:
I sort of like how the other checkpoint bot calculates the poll interval so that it always targets a particular time in the checkpoint (like halfway through the checkpoint) to checkpoint. It only needs to check once a checkpoint if we use this method. The method used here definitely works, but it will result in the checkpoint bots minting more of the checkpoints for lower volume pools. The downside of this is that it costs money to mint a checkpoint, and it would be preferable to wait until some amount of the checkpoint has passed before re-minting a checkpoint. 

I say all of this to add some context for why the other checkpoint bot works the way that it does. These concerns aren't relevant for the mainnet fork environment, but we'd most likely want to make these changes if we decided to use this bot script for mainnet.

Reply:
Really appreciate the context here, thank you so much!

Will leave as is since this script is ideally temporary until agent0 checkpointing gets used.